tools/distrib/check_redundant_namespace_qualifiers.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python
+# Copyright 2021 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Eliminate the kind of redundant namespace qualifiers that tend to
+# creep in when converting C to C++.
+
+import collections
+import os
+import re
+import sys
+
+
+def find_closing_mustache(contents, initial_depth):
```

Comment:
I don't suppose there's a clang-tidy check for this, is there?  It would be nice to have a cleaner way to do this that this kind of bespoke parsing logic, just because I know that we'll wind up running into an edge case at some point. :/

Reply:
I checked, and nope...

Reply:
And so we mustache whomever finds the edge-case to update things here.